src/lib/qgram.rs

```diff
@@ -83,15 +83,44 @@ impl QGram {
         distance
     }
 
+    /// Choose a padding character that does not appear in `input`.
+    ///
+    /// Use `self.padding` if possible.
+    /// Panics if it cannot find a suitable padding character.
+    fn choose_padding(&self, input: &str) -> String {
+        // Note that `first()` is not defined for the `Chars` type.
+        if !input.contains(self.padding.chars().nth(0).unwrap()) {
```

Comment:
This makes this function O(2n) when we can make this O(n) by folding this into the `for` loop below and having a flag `found_padding_char` which is set to true if `c == padding_char`. [Hopefully that makes sense.]